### PR TITLE
Add unit test for JSONArray

### PR DESCRIPTION
**Pull request by 3mmar7amed**
## Purpose / Description
Add unit test for JSONArray class.
this class has not been tested yet.

## Approach
comparing the value stored in a specific index in a list with the actual value.

## How Has This Been Tested?
Passed the unit test locally.

## Checklist
_Please, go through these checks before submitting the PR._

- [✔ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ✔ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [✔  ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [✔  ] You have commented your code, particularly in hard-to-understand areas
- [ ✔ ] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.ichi2.utils;
+import org.junit.jupiter.api.Assertions;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.java:3]: We typically use junit `Assert.assertEquals`, or Hamcrest: no particular reason, but it's nice to stick to a convention within the codebase

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
A few style-based nitpicks

But, the main question is "What are we testing".

If I'm testing `put`, I'd typically want a few tests testing different things:

* Test putting into an empty array
* Test inserting at the start of a filled array
* Ditto inserting at the middle and end
* Ensure that the length increases if put is called
* Test putting null (if possible)
* If there's a return value, what is it? Is the original array modified, or is it immutable and returning a new instance
* (Possibly) boundary checks: put -1, or past the end and see what happens: if past the end, does it fail, or is the array expanded

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
It doesn't look like there were changes since my previous review

**Comment by david-allison-1** [comment]
@3mmar7amed Is this superseded by #8524 ?

**Comment by 3mmar7amed** [comment]
> @3mmar7amed Is this superseded by #8524 ?

yes

**Pull request closed without merging**